ambry-store/src/main/java/com/github/ambry/store/CompactionManager.java

```diff
@@ -288,6 +288,9 @@ public void run() {
                     metrics.markCompactionStart(true);
                     compactionStarted = true;
                     store.compact(details, bundleReadBuffer);
+                    if (store.getReplicaId().isSealed()) {
```

Comment:
why not add config `storeAutoCloseLastLogSegmentEnabled` check in this level(top level) to control auto close?

Reply:
done.